circuit_maintenance_parser/provider.py

```diff
@@ -59,12 +69,44 @@ class GenericProvider(BaseModel):
     _processors: List[GenericProcessor] = [SimpleProcessor(data_parsers=[ICal])]
     _default_organizer: str = "unknown"
 
+    _include_filter: Dict[str, List[str]] = {}
+    _exclude_filter: Dict[str, List[str]] = {}
+
+    def include_filter_check(self, data: NotificationData) -> bool:
+        """If `_include_filter` is defined, it verifies that the matching criteria is met."""
+        if self._include_filter:
+            return self.filter_check(self._include_filter, data)
+        return True
+
+    def exclude_filter_check(self, data: NotificationData) -> bool:
+        """If `_exclude_filter` is defined, it verifies that the matching criteria is met."""
+        if self._exclude_filter:
+            return self.filter_check(self._exclude_filter, data)
+        return False
+
+    @staticmethod
+    def filter_check(filter_dict: Dict, data: NotificationData) -> bool:
+        """Generic filter check."""
+        for data_part in data.data_parts:
```

Comment:
I wonder - should we log a warning or something if we have a data type in `filter_dict` that isn't present in `data`? I.e. if we include based on `email-header-subject` but the data we're processing is missing that field?

Reply:
yes, I was not logging enough, will improve :)